src/App.tsx

```diff
@@ -1,24 +1,74 @@
-/* eslint-disable max-len */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useCallback, useContext, useEffect } from 'react';
 
-const USER_ID = 0;
+import { DispatchContext, StateContext } from './Store';
+
+import { getTodos } from './api/todos';
+
+import { TodoHeader } from './components/TodoHeader';
+import { TodoList } from './components/TodoList';
+import { TodoFooter } from './components/TodoFooter';
+import { ErrorNotification } from './components/ErrorNotification';
+
+import { Error } from './types/Error';
+import { Todo } from './types/Todo';
+import { Filter, FilterTitles } from './types/Filter';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const dispatch = useContext(DispatchContext);
+  const { todos, filter } = useContext(StateContext);
+
+  const filterTodos = useCallback((
+    todosToFilter: Todo[],
+    currentFilter: Filter,
+  ) => {
+    switch (currentFilter.title) {
+      case FilterTitles.All:
+      default:
+        return todosToFilter;
+
+      case FilterTitles.Active:
+        return todosToFilter.filter(todo => !todo.completed);
+
+      case FilterTitles.Completed:
+        return todosToFilter.filter(todo => todo.completed);
+    }
+  }, []);
+
+  const handleLoadTodosError = useCallback(() => {
+    dispatch({ type: 'setError', payload: Error.LoadTodosError });
+    setTimeout(() => {
+      dispatch({ type: 'setError', payload: '' });
+    }, 3000);
+  }, [dispatch]);
+
+  useEffect(() => {
+    dispatch({ type: 'setError', payload: '' });
+    getTodos()
+      .then(todosFromServer => {
+        dispatch({
+          type: 'setTodos',
+          payload: todosFromServer,
+        });
+      })
+      .catch(handleLoadTodosError);
+  }, [dispatch, handleLoadTodosError]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <TodoHeader />
+
+        <TodoList todos={filterTodos(todos, filter)} />
+
+        {todos.length > 0 && (
```

Comment:
```suggestion
        {!!todos.length && (
```
**PLS DO THE SAME FOR ALL SUCH CASES IN THE PROJECT**